notes_app/notes/tests.py

```diff
@@ -155,3 +155,291 @@ def test_delete_detail(self):
+        original_data = self.deserialize(
+            self.api_client.get(
+                self.api_detail,
+                format='json',
+                authentication=self.get_credentials()
+            )
+        )
+        new_data = original_data.copy()
+        new_data['first_name'] = 'new First'
+        new_data['last_name'] = 'new Last'
+        total_count = User.objects.all().count()
+
+        self.assertEqual(User.objects.count(), total_count)
+        self.assertHttpAccepted(
+            self.api_client.put(
+                self.api_detail,
+                format='json',
+                data=new_data,
+                authentication=self.get_credentials()
+            )
+        )
+        self.assertEqual(User.objects.count(), total_count)
+        self.assertEqual(User.objects.get(id=self.api_user_id).first_name, 'new First')
+        self.assertEqual(User.objects.get(id=self.api_user_id).last_name, 'new Last')
+
+    def test_put_detail_auth_fail(self):
+        """Users that fail Authorization can't PUT"""
+        original_data = self.deserialize(
+            self.api_client.get(
+                self.api_detail,
+                format='json',
+                authentication=self.get_credentials()
+            )
+        )
+        original_fname = original_data['first_name']
+        original_lname = original_data['last_name']
+        new_data = original_data.copy()
+        new_data['first_name'] = 'new First'
+        new_data['last_name'] = 'new Last'
+        total_count = User.objects.all().count()
+
+        self.assertEqual(User.objects.count(), total_count)
+        self.assertHttpUnauthorized(
+            self.api_client.put(
+                self.api_detail,
+                format='json',
+                data=new_data,
+                authentication=self.get_wrong_credentials()
+            )
+        )
+        self.assertEqual(User.objects.count(), total_count)
+        self.assertNotEqual(User.objects.get(id=self.api_user_id).first_name, 'new First')
+        self.assertEqual(User.objects.get(id=self.api_user_id).first_name, original_fname)
+        self.assertNotEqual(User.objects.get(id=self.api_user_id).last_name, 'new Last')
+        self.assertEqual(User.objects.get(id=self.api_user_id).last_name, original_lname)
+
+    def test_patch_detail(self):
+        """Authorized User can PATCH"""
+        resp = self.api_client.put(
+            self.api_detail,
+            format='json',
+            data=self.patch_data,
+            authentication=self.get_credentials()
+        )
+        self.assertHttpAccepted(resp)
+        self.assertEqual(User.objects.get(id=self.api_user_id).last_name, 'testPatch')
+
+    def test_patch_detail_auth_fail(self):
+        """Users that fail Authorization can't PATCH"""
+        resp = self.api_client.put(
+            self.api_detail,
+            format='json',
+            data=self.patch_data,
+            authentication=self.get_wrong_credentials()
```

Comment:
it looks like you've got me wrong. By testing authorization I meant `UserAuthorization` class logic.